pytorch_transformers/tokenization_roberta.py

```diff
@@ -12,229 +12,182 @@
-        return self.unk_index
-
-    def add_symbol(self, word, n=1):
-        """Adds a word to the dictionary"""
-        if word in self.indices:
-            idx = self.indices[word]
-            self.count[idx] = self.count[idx] + n
-            return idx
-        else:
-            idx = len(self.symbols)
-            self.indices[word] = idx
-            self.symbols.append(word)
-            self.count.append(n)
-            return idx
-
-    @classmethod
-    def load(cls, f, ignore_utf_errors=False):
-        """Loads the dictionary from a text file with the format:
-
-        ```
-        <symbol0> <count0>
-        <symbol1> <count1>
-        ...
-        ```
-        """
-        d = cls()
-        d.add_from_file(f, ignore_utf_errors)
-        return d
-
-    def add_from_file(self, f, ignore_utf_errors=False):
-        """
-        Loads a pre-existing dictionary from a text file and adds its symbols
-        to this instance.
-        """
-        if isinstance(f, six.string_types):
-            try:
-                if not ignore_utf_errors:
-                    with open(f, 'r', encoding='utf-8') as fd:
-                        self.add_from_file(fd)
-                else:
-                    with open(f, 'r', encoding='utf-8', errors='ignore') as fd:
-                        self.add_from_file(fd)
-            except FileNotFoundError as fnfe:
-                raise fnfe
-            except UnicodeError:
-                raise Exception("Incorrect encoding detected in {}, please "
-                                "rebuild the dataset".format(f))
-            return
-
-        lines = f.read().splitlines()
-        for line in lines:
-            idx = line.rfind(' ')
-            if idx == -1:
-                raise ValueError("Incorrect dictionary format, expected '<token> <cnt>'")
-            word = line[:idx]
-            count = int(line[idx + 1:])
-            self.indices[word] = len(self.symbols)
-            self.symbols.append(word)
-            self.count.append(count)
-
-    def encode_line(self, line, line_tokenizer=tokenize_line, add_if_not_exist=True,
-                    consumer=None, append_eos=True, reverse_order=False):
-        words = line_tokenizer(line)
-        if reverse_order:
-            words = list(reversed(words))
-        nwords = len(words)
-        ids = [0] * (nwords + 1 if append_eos else nwords)
-
-        for i, word in enumerate(words):
-            if add_if_not_exist:
-                idx = self.add_symbol(word)
-            else:
-                idx = self.index(word)
-            if consumer is not None:
-                consumer(word, idx)
-            ids[i] = idx
-        if append_eos:
-            ids[nwords] = self.eos_index
-        return ids
-
 
 class RobertaTokenizer(PreTrainedTokenizer):
     """
-    RoBERTa tokenizer. Peculiarities:
-        - GPT-2 tokenizer with a different integer mapping on top.
+    GPT-2 BPE tokenizer. Peculiarities:
```

Comment:
You can keep "RoBERTa" here and just mention that it is derived from GPT-2.
We should actually have better docstrings for the Tokenizer, docstrings that reference the papers, like for the models.